src/algorithm/hnswlib/hnswalg.cpp

```diff
@@ -73,6 +71,10 @@ HierarchicalNSW::HierarchicalNSW(SpaceInterface* s,
 
 void
 HierarchicalNSW::reset() {
+    if (visited_list_pool_) {
+        VisitedListPool::operator delete(visited_list_pool_, allocator_);
```

Comment:
delete visitored_list_pool_

Reply:
Because an allocator is used, the default new and delete cannot be used.

Reply:
OK, I have modified it.